### ISSUE TITLE
Show arrays as additional rows instead of columns

### ISSUE THREAD

**Issue by cedeerwe** [wildmountainfarms/wild-graphql-datasource#6]
First of all, thanks for the great plugin, the idea with embedding Graphiql is very neat!

Example I am using: https://countries.trevorblades.com/
Query:
```
query {
  countries {
    name,
    languages {
      name
    }
  }
}
```
Data Path: countries
Visualization: Table

I am getting a table with many columns, one for each index of the languages array. A much more intuitive and usable representation would be to have the first column repeated and the language column changing, i.e. instead of this:

```
+-------------+-----------------+-----------------+-----------------+
| name        | language.0.name | language.1.name | language.2.name |
+-------------+-----------------+-----------------+-----------------+
| Andorra     | Catalan         |                 |                 |
+-------------+-----------------+-----------------+-----------------+
| Afghanistan | Pashto          | Uzbek           | Turkmen         |
+-------------+-----------------+-----------------+-----------------+
```

To have this:

```
+-------------+---------------+
| name        | language.name |
+-------------+---------------+
| Andorra     | Catalan       |
+-------------+---------------+
| Afghanistan | Pashto        |
+-------------+---------------+
| Afghanistan | Uzbek         |
+-------------+---------------+
| Afghanistan | Turkmen       |
+-------------+---------------+
```

I understand that this might not be everybody's need (mine is 😂) and therefore having this as an optional parsing method for arrays would also make sense.

WDYT?

**Comment by retrodaredevil** [comment]
Yeah I agree this would be useful. I'll have to think of the best way to implement this. Maybe the query could have a list of "expand array" paths that would expand the array like you've shown. 

Not sure when I'll be able to get around to this, but it will be on my to-do list.

**Comment by cedeerwe** [comment]
Would you have an honest estimate of when this feature could land? Or should I have a shot at implementing it myself, if I would like it in the upcoming weeks?

**Comment by retrodaredevil** [comment]
I have no estimate. Probably best for you to implement it yourself if you want it in the next few weeks.

**Comment by Danile71** [comment]
I am trying to solve a issue.

https://github.com/wildmountainfarms/wild-graphql-datasource/pull/13

**Comment by retrodaredevil** [comment]
@Danile71 I left some comments on #13. I still need to take a closer look at it. Once I take a closer look I might be able to recommend a solution for how we could best go about solving this problem.

Before you make too much progress on this, I'd also like to say that the solution should still support the old way. Maybe a toggle between the "flat way" (how it is currently), the "additional rows way" (what is proposed in the OP of this issue), and then maybe also a "nested way"?

It looks like you were going for some sort of nested way such that the value of one of the fields in the data frame is actually an array. I don't think that will be useful for anything within Grafana. It's my understanding that values within a data frame need to be primitive values and should not be objects or arrays. I could be wrong, but that was my understanding and my interpretation of what you were trying to do.

**Comment by retrodaredevil** [comment]
I'm hoping to spend some more time working on Wild GraphQL Datasource in the next couple of weeks, but it likely won't be to address this issue. Testing changes like this can be difficult. I need to put some time into the provisioned dashboards so we can make changes and be confident they won't break anything.

I was hoping I could be more active on this project, but I'm doing my best :)

**Comment by retrodaredevil** [comment]
I'm gonna do a little brainstorming session for myself here:


# My Use Case

```json
{
  "dataIDontCareAbout": "asdf",
  "response": [
    {
      "dateMillis": 1234,
      "packet": {
        "batteryVoltage": 24.3
      }
    }
  ]
}
```

This is what is currently supported. The data path would be `response`, time path would be `dateMillis` and each entry would have `dateMillis` and `packet.batteryVoltage` as the field names.

# Advanced Use Case

```json
{
  "serverName": "Central Server 1",
  "response": {
    "data": [
      {
        "node": "node 1",
        "data": [
          {
            "dateMillis": 1234,
            "data": [
              { "processor": 0, "temperature": 30.2 },
              { "processor": 1, "temperature": 31.0 },
            ]
          }
        ]
      },
      {
        "node": "node  2",
        "data": [
          {
            "dateMillis": 1234,
            "data": [
              { "processor": 0, "temperature": 36.2 },
              { "processor": 1, "temperature": 35.0 },
            ]
          }
        ]
      }
    ]
  }
}
```

With this blob of JSON data, configuring it won't be as trivial, but the goal is to have these fields available to us:

* `serverName`
* `response.data.node`
* `response.data.data.dateMillis`
* `response.data.data.data.processor`
* `response.data.data.data.temperature`

In our case, the data path would be empty, and the time path would be `response.data.data.dateMillis`. Additionally, we would want the "expand to long format paths" to be `response.data`, `response.data.data`, `response.data.data.data`, which could probably just be simplified to `response.data.data.data`.

The end result should be something like

| serverName     | response.data.node | response.data.data.dateMillis | response.data.data.data.processor | response.data.data.data.temperature |
|----------------|--------------------|-------------------------------|-----------------------------------|-------------------------------------|
| Central Server 1 | node 1            | 1234                          | 0                                 | 30.2                                |
| Central Server 1 | node 1            | 1234                          | 1                                 | 31.0                                |
| Central Server 1 | node 2            | 1234                          | 0                                 | 36.2                                |
| Central Server 1 | node 2            | 1234                          | 1                                 | 35.0                                |

As I mentioned in the previous comment, we don't want to break existing functionality, so if the "expand to long format paths" is not specified for a given array, then it would default to the old way.

This "expand to long format paths" would have to be one of the first things applied, likely being applied right after the data path is applied. I'll have to check to make sure that we support the data path being empty either way.

**Comment by retrodaredevil** [comment]
This feature will be available in the 1.4.0 release
